opentasks-provider/src/androidTest/java/org/dmfs/provider/tasks/TaskProviderTest.java

```diff
@@ -199,7 +232,111 @@ public void testInsertTaskWithStartAndDue()
                 new Assert<>(task, new TimeData(start, due)),
                 new AssertRelated<>(new InstanceTable(mAuthority), Instances.TASK_ID, task, new AllOf(
                         new EqArg(Instances.INSTANCE_START, start.getTimestamp()),
+                        new EqArg(Instances.INSTANCE_ORIGINAL_TIME, start.getTimestamp()),
+                        new EqArg(Instances.INSTANCE_DUE, due.getTimestamp()),
+                        new EqArg(Instances.INSTANCE_START_SORTING, start.getInstance()),
+                        new EqArg(Instances.INSTANCE_DUE_SORTING, due.getInstance()),
+                        new EqArg(Instances.INSTANCE_DURATION, due.getTimestamp() - start.getTimestamp()),
+                        new EqArg(Tasks.TZ, start.isAllDay() ? "UTC" : start.getTimeZone().getID())
+                ))
+        ));
+    }
+
+
+    /**
+     * Create task with start and due and update it with new values, check datetime values including generated duration.
+     */
+    @Test
+    public void testInsertTaskWithStartAndDueMovedForward()
```

Comment:
About 2/3 of the tests fail for me, including this one, with:
`java.lang.AssertionError: android.content.OperationApplicationException: wrong number of rows: 0`

Do they pass for you?

Reply:
They pass for me and for Travis. If they don't pass for you, there might be an issue with the database migration code. It's either not executed or incorrect. I'll check that.

Reply:
btw, If you uninstall the test project from your device, the tests should pass on the next launch. This would clearly indicate a migration issue.

Reply:
Tests pass now after the fix, on non-UTC timezone device, too.

Created ticket to try to avoid this in the future: https://github.com/dmfs/opentasks/issues/571